File: packages/java/src/com/peruspatial/territorial/CalculateTerritorialRisk.java
package com.peruspatial.territorial;

import java.util.List;

public final class CalculateTerritorialRisk {
  private CalculateTerritorialRisk() {}

  public static RiskResult calculateTerritorialRisk(List<Intersection> intersections) {
    if (intersections == null || intersections.isEmpty()) {
      return new RiskResult("BAJO", 0);
    }
    int sum = 0;
    for (Intersection i : intersections) {
      sum += weight(i.level());
    }
    int score = Math.min(100, sum);
    String risk = score >= 67 ? "ALTO" : score >= 34 ? "MEDIO" : "BAJO";
    return new RiskResult(risk, score);
  }

  private static int weight(String level) {
    if (level == null) return 0;
    return switch (level) {
      case "BAJO" -> 10;
      case "MEDIO" -> 20;
      case "ALTO" -> 34;
      default -> 0;
    };
  }
}

File: packages/go/calculate_territorial_risk.go
package perspatial

func CalculateTerritorialRisk(intersections []Intersection) RiskResult {
	if len(intersections) == 0 {
		return RiskResult{Risk: "BAJO", Score: 0}
	}
	sum := 0
	for _, in := range intersections {
		sum += levelWeight(in.Level)
	}
	if sum > 100 {
		sum = 100
	}
	risk := "BAJO"
	switch {
	case sum >= 67:
		risk = "ALTO"
	case sum >= 34:
		risk = "MEDIO"
	}
	return RiskResult{Risk: risk, Score: sum}
}

func levelWeight(level string) int {
	switch level {
	case "BAJO":
		return 10
	case "MEDIO":
		return 20
	case "ALTO":
		return 34
	default:
		return 0
	}
}
